### DISCUSSION TITLE
Java Rest API 404error

### DISCUSSION THREAD

**Discussion by setapex** [vercel/community#2053] [Help] [answered]
### Summary

I have a small project on java rest, i deployed it to vercel. At local it works correctly but on web i have 404error. U can check my rep tihomirka, I will be grateful for your help

**Comment by amyegan** [comment, marked as answer]
Hi @setapex! Java is not currently a supported runtime on Vercel. 

We currently have [serverless function runtimes](https://vercel.com/docs/concepts/functions/serverless-functions/runtimes#) available for [Node.js](https://vercel.com/docs/concepts/functions/serverless-functions/runtimes/node-js) (JavaScript), [Go](https://vercel.com/docs/concepts/functions/serverless-functions/runtimes/go), [Python](https://vercel.com/docs/concepts/functions/serverless-functions/runtimes/python), and [Ruby](https://vercel.com/docs/concepts/functions/serverless-functions/runtimes/ruby). There are also [community maintained runtimes](https://vercel.com/docs/concepts/functions/serverless-functions/runtimes#community-runtimes) available for [Bash](https://github.com/importpw/vercel-bash), [Deno](https://github.com/vercel-community/deno), [PHP](https://github.com/vercel-community/php), and [Rust](https://github.com/vercel-community/rust)
